lib/src/modules/network_logger.dart

```diff
@@ -66,13 +70,88 @@ class NetworkLogger {
   }
 
   Future<void> networkLog(NetworkData data) async {
-    final omit = await _manager.omitLog(data);
-
+    final networkData = await _addW3Header(data);
+    final omit = await _manager.omitLog(networkData);
     if (omit) return;
-
-    final obfuscated = await _manager.obfuscateLog(data);
-
+    final obfuscated = await _manager.obfuscateLog(networkData);
     await _host.networkLog(obfuscated.toJson());
     await APM.networkLogAndroid(obfuscated);
   }
+
+  Future<NetworkData> _addW3Header(NetworkData data) async {
```

Comment:
Let's use the actual name and change W3* members to W3C*, for example, changing `_addW3Header` to `_addW3CHeader`, to maintain consistency and descriptiveness.